File: app/models/comment.rb
class Comment < ActiveRecord::Base
  include ActsAsCommentable::Comment

  belongs_to :commentable, :polymorphic => true

  default_scope :order => 'created_at ASC'

  belongs_to :user

  def to_hash
    {
      text: comment,
      created_at: created_at.to_s(:short),
      location_id: commentable_id,
      user: user.login,
      thumb: "http://www.gravatar.com/avatar/#{Digest::MD5.hexdigest(user.login)}?d=identicon&s=25"
    }
  end

end

File: app/assets/javascripts/models/comment.js
var Comment = Model("comment", function() {
  this.extend({
    find_by_location_id: function(location_id) {
      return this.select(function() {
        return this.attr("location_id") == parseInt(location_id)
      })
    }
  });
  this.include({
    to_hash: function() {
      return {user: this.attr("user"), text: this.attr("text"), created_at: this.attr("created_at")};
    }
  });
});